HaloDAO/lending-market-v1
# Adding new asset, fixes

## Description of Changes


Fixed local deploy errors
Added tasks for adding new asset
[Link to Jira Ticket](https://halodao.atlassian.net/browse/LEND-4)

## How To Test

#### Testing local deploy:
1. In one terminal: `yarn hardhat node`
2. In another terminal, generate typechain types first by running `yarn hardhat compile`
3. You can now deploy the whole environment in your local node. Run `yarn hardhat halo:dev --withmocktokens true` (this installs mock tokens for local dev environment, if you want to read from your previously generated `deployed-contracts.json` file then set this to false)

#### Testing kovan deploy:
1. If you made any changes from the smart contracts, make sure to recompile them first since some deplyment scripts skips compiling. Run `yarn recompile`
2. Make sure you have the updated Kovan assets in the `markets` directory on kovan
3. Deploy the whole environment to Kovan by running `yarn hardhat halo:dev --withmocktokens false --network kovan`
4. Test using the tasks in `Lending Pool Tasks`

#### Testing adding a new asset - ERC20 token
1. Make sure you are using the forked version of hardhat config. Refer to the comments on `./hardhat.config.ts` in the `buidlerConfig.networks.hardhat` with blockNumber set at `14632357` (before deploying XSGD reserve on mainnet)
2. Try XSGD in the mainnet fork. Run `yarn hardhat halo:new-asset --symbol XSGD`


#### Testing adding a new asset - LP Token
1. Make sure you are using the forked version of hardhat config. Refer to the comments on `./hardhat.config.ts` in the `buidlerConfig.networks.hardhat` with blockNumber set at `14632357`
2. Try HLP_XSGD_USDC in the mainnet fork. Run `yarn hardhat halo:new-asset --symbol HLP_XSGD_USDC  --lp` (before deploying HLP_XSGD_USDC reserve on mainnet)

#### Testing disable borrow 
1. Make sure you are using the forked version of hardhat config. Refer to the comments on `./hardhat.config.ts` in the `buidlerConfig.networks.hardhat` with blockNumber set at `14651807` or later (after XSGD and HLP_XSGD_USDC assets were initialized)
2. Run `yarn hardhat external:disable-borrow-reserve --symbol  HLP_XSGD_USDC --lp`


#### Running the test suite
You can run the full test suite with the following commands:

```
yarn test
```

NOTES:

- You can use a non forked version of the hardhat node to run this in an isolated environment (which is preferred). Refer to the comments on `./hardhat.config.ts` in the `buidlerConfig.networks.hardhat`
- Test may fail when the test engine runs incentives controller tests. Sometimes its off by 1. Just run the test again.


## Developer Checklist:


### Reviewers Checklist:
